tasks/ast_tools/src/generators/ast_kind.rs

```diff
@@ -154,6 +155,23 @@ impl Generator for AstKindGenerator {
             .map(|(ident, _)| parse_quote!(Self :: #ident(it) => it.span()))
             .collect_vec();
 
+        let as_ast_kind_impls: Vec<ImplItemFn> = have_kinds
+            .iter()
+            .map(|(ident, typ)| {
+                let snake_case_name =
+                    format_ident!("as_{}", ident.to_string().to_case(Case::Snake));
+                parse_quote!(
+                    pub fn #snake_case_name(&self) -> Option<&'a #typ> {
```

Comment:
Please add `///@@line_break` before this line to insert line breaks into output, making the generated code easier to read.

(`///@@line_break` is a workaround for that `parse_quote!` and `quote!` macros do not take any notice of whitespace)